examples/Cpp/PrimitiveRestart/Example.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Example.cpp (Example_PrimitiveRestart)
+ *
+ * This file is part of the "LLGL" project (Copyright (c) 2015-2019 by Lukas Hermanns)
+ * See "LICENSE.txt" for license information.
+ */
+
+#include <ExampleBase.h>
+#include <vector>
+
+class Example_PrimitiveRestart : public ExampleBase
+{
+    LLGL::ShaderProgram*    shaderProgram       = nullptr;
+    LLGL::PipelineState*    pipeline            = nullptr;
+    LLGL::Buffer*           vertexBuffer        = nullptr;
+    LLGL::Buffer*           indexBuffer16       = nullptr;
+    LLGL::Buffer*           indexBuffer32       = nullptr;
+
+    size_t                  indexCount16        = 0;
+    size_t                  indexCount32        = 0;
+
+public:
+
+    Example_PrimitiveRestart() :
+    ExampleBase { L"LLGL Example: PrimitiveRestart" }
+    {
+        auto vertexFormats = CreateBuffers();
+        shaderProgram = LoadStandardShaderProgram(vertexFormats);
+        CreatePipeline();
+    }
+
+private:
+
+    // Vertex data structure
+    struct Vertex
+    {
+        Gs::Vector2f        position;
+        LLGL::ColorRGBAub   color;
+    };
+
+    template <typename T>
+    static void addSquare(float centerX, float centerY, float size, std::vector<Vertex>& vertices, std::vector<T>& indices)
+    {
+        float left = centerX - size / 2.0;
+        float right = centerX + size / 2.0;
+        float top = centerY + size / 2.0;
+        float bottom = centerY - size / 2.0;
+
+        size_t startIndex = vertices.size();
+        vertices.push_back({{ right, top    }, { 255,   0,   0, 255 } });
+        vertices.push_back({{ right, bottom }, {   0, 255,   0, 255 } });
+        vertices.push_back({{ left,  top    }, {   0,   0, 255, 255 } });
+        vertices.push_back({{ left,  bottom }, { 255, 255, 255, 255 } });
+
+        if (!indices.empty())
+            indices.push_back((T) 0xFFFFFFFF);
+        indices.push_back(startIndex);
+        indices.push_back(startIndex + 1);
+        indices.push_back(startIndex + 2);
+        indices.push_back(startIndex + 3);
+    }
+
+    std::vector<LLGL::VertexFormat> CreateBuffers()
+    {
+        // Vertex data, an array of 16 points that we'll use
+        // to draw triangles.
+        const float s = 0.5f;
+
+        std::vector<Vertex> vertices;
+        std::vector<uint16_t> indices16;
+        std::vector<uint32_t> indices32;
+
+        addSquare( .5,  .5, 0.8, vertices, indices16);
```

Comment:
[MSVC emits warnings here](https://ci.appveyor.com/project/LukasBanana/llgl/builds/34031207/job/q1mfig2h0l6a5umd#L191) because of `double` literals used for `float` parameters in a template. Simply adding the `f` suffix should fix that.